modules/ROOT/partials/configuration/text_patterns.adoc

```diff
@@ -0,0 +1,148 @@
+[[text_patterns]]
+== `+text_patterns+`
+
+This option allows configuring the text patterns that get matched while typing in the editor. Text patterns can be used to apply formats, replace text, or execute commands. By default, basic markdown patterns are enabled so the user can type `+# text+` to produce a header or `+**text**+` to make text *bold*. To disable text patterns getting replaced, set the option to `false`.
+
+There are three types of patterns: `+inline+`, `+block+`, and `+replacement+` patterns. Inline patterns have a `+start+` and an `+end+` text pattern whereas the block and replacement patterns only have a `+start+`. A user can specify the formats to be applied to the selection, commands to be executed, or text to be replaced.
+
+[IMPORTANT]
+====
+Any formats or commands used in text patterns need to be registered with the editor when it is initialized. This may include enabling relevant plugins, such as the `+lists+` plugin. For information on:
+
+* Registering formats for {productname}, see: xref:content-formatting.adoc#formats[Content formatting options - `+formats+`].
+* Registering commands for {productname}, see: xref:apis/tinymce.editor.adoc#addCommand[{productname} APIs - addCommand].
+====
+
+Type : `+Array+` or `+false+`
+
+Default Value:
+[source,js]
+----
+[
+  { start: '*', end: '*', format: 'italic' },
+  { start: '**', end: '**', format: 'bold' },
+  { start: '#', format: 'h1' },
+  { start: '##', format: 'h2' },
+  { start: '###', format: 'h3' },
+  { start: '####', format: 'h4' },
+  { start: '#####', format: 'h5' },
+  { start: '######', format: 'h6' },
+  // The following text patterns require the `lists` plugin
+  { start: '1. ', cmd: 'InsertOrderedList' },
+  { start: '* ', cmd: 'InsertUnorderedList' },
+  { start: '- ', cmd: 'InsertUnorderedList' }
+]
+----
+
+=== Inline patterns
+
+Inline patterns must have the following:
+
+* A `+start+` and an `+end+`.
+* A `+format+` or a `+cmd+`.
+** If `+cmd+` is specified, an optional `+value+` property is allowed.
+
+This allows for patterns to be used to either apply a format or execute a command, optionally with the given value.
+
+NOTE: Inline patterns are executed on either pressing the *spacebar* or the *Enter* key.
+
+==== Example: Using inline patterns
+
+[source,js]
+----
+tinymce.init({
+  selector: 'textarea',  // change this value according to your HTML
+  // The `link` plugin is required for the `createLink` command
+  plugin: 'link',
+  text_patterns: [
+    { start: '*', end: '*', format: 'italic' },
+    { start: '**', end: '**', format: 'bold' },
+    { start: '~', end: '~', cmd: 'createLink', value: 'https://tiny.cloud'}
```

Comment:
```suggestion
    { start: '~', end: '~', cmd: 'createLink', value: 'https://tiny.cloud' }
```
It looks like there are also a few other places where a space is missing before the closing bracket

Reply:
Thanks, it seems my find & replace regex missed a couple of cases 😞  I believe I've got all of them in this file now.